### ISSUE TITLE
Get-LSUpdate -All without IsApplicable and IsInstalled Checks (execution)

### ISSUE THREAD

**Issue by bechtlepri** [jantari/LSUClient#21]
**Describe the solution you'd like**
When the `-All` switch is used, the check for installed packages (with installer/executables) should be optional.
Maybe add a `-CheckInstalled:$false` and `-CheckApplicable:$false` argument to have more control?

**Describe alternatives you've considered**
Using a fork of the project with the change.

**Additional context**
We use the script beforehand to download all packages for a specific model (with -All and -Model) on our development machines and not the target systems.
In this case, we neither want the LSUClient to check for installed software nor execute any installer/executables.
Also, all packages are considered to be "applicable".

In our case, we only want to Get (and save the JSON for) all available packages for a specific model.

See the following PR for a suggestion.

**Comment by jantari** [comment]
Hi, thank you for the issue and PR.

I agree there should be a way to skip these checks, however I think there is also a use case for retrieving all packages AND keeping the IsInstalled and IsApplicable tests like it is now - for example:

- You want to generally only install the needed packages, except for wifi drivers which you always want to (re-)install
- If you know a package is often falsely detected as installed even though it isn't

This is an example I mention in the readme:

```powershell
Get-LSUpdate -All | Where-Object { $_.IsApplicable -and -not $_.IsInstalled }
```

to retrieve all packages, but filter them later according to your own custom logic. I also use this to help test the module, to find packages that are being detected wrong.

I think we need extra parameters like you suggested, probably in a parameterset with `-All` because it wouldn't make sense to use e.g. `-CheckApplicable:$false` _without_ `-All`

**Comment by jantari** [comment]
Hi, I pushed an initial implementation of this to the [develop branch](/jantari/LSUClient/tree/develop) from where you can download and try it out.

I decided to make the `.IsApplicable` and `.IsInstalled` properties nullable bools, and to leave/set them to `$null` when you are skipping the checks - because "wasn't checked" is different from e.g. "it's not installed" and I want to make that difference clear in the returned objects.

So .IsApplicable and .IsInstalled can now be true, false or null.

Also, the new parameters I added are `-NoTestApplicable` and `-NoTestInstalled`. They are dynamic parameters that only show up and work once you've specified `-All`. You can test this in the PowerShell console through tab-completion. I did this because the default behavior of the module is, like the official System Update, to only show you updates your system is missing. If a user were able to specify `-NoTestSomething` without `-All`, I really wouldn't know what do do with the package - do I show it or not? I prefer when you have to explicitly say `-All` as well to be able to skip the checks.

I decided on the (No)Test verb because unlike 'check' it's a PowerShell approved verb, so fits better into the language I feel.

Would like your feedback on this though!

**Comment by bechtlepri** [comment]
Hey, thanks a lot for the update / work. Looking forward seeing the next release.

All sounds good to me
**One thing. though:**
I do not recommend _"dynamic parameters that only show up and work once you've specified `-All`"_ because it is a bit confusing. I agree, that you must not be able to use the new switches without `-All` but I would prefer an error on start than no auto complete.

**Reason:**
Usually, when I use a new command I just tab through all possible options and "try them out". I would not do this after specifying `-Al`l again and probably not see the `-NoTest*`.

But it's a minor issue, as soon as you check the documentation you'll see the parameters anyways.
So I am good either way.

**Comment by jantari** [comment]
Yea I get your point, I too use auto-complete to discovver parameters first.

Plus the dynamic parameters add a kinda ugly code block to the function, so I'll think about it.

**Comment by jantari** [comment]
The new parameters now shipped with Version 1.2.0
